src/Form/Control/Password.php

```diff
@@ -4,7 +4,46 @@
 
 namespace Atk4\Ui\Form\Control;
 
+use Atk4\Ui\Button;
+use Atk4\Ui\Js\JsExpression;
+
 class Password extends Line
 {
     public string $inputType = 'password';
+
+    /** By setting this variable to true you get an Eye-button on the right side which can toggle the visibility of the entered password */
+    public bool $eye;
+
+    #[\Override]
+    protected function init(): void
+    {
+        parent::init();
+
+        if ($this->eye) {
+            $button = new Button([
+                'class.grey' => true,
+                'iconRight' => 'eye slash',
+            ]);
+
+            $this->action = $button;
+
+            $button->on(
+                'click',
+                new JsExpression(
+                    <<<EOF
```

Comment:
```suggestion
                    <<<'EOF'
```

and pass all strings using JsExpression templation to prevent JS injections

Reply:
It's all good and you will have to guide me... :)

Anyway, yes,
1, getElementById works for the input field, how can I get the ID of the icon connected to the same field, it has a different ID.

I think I need more information on a lot of things, like Behat for exmple.

Reply:
> and pass all strings using JsExpression templation to prevent JS injections

see https://github.com/atk4/ui/blob/5.2.0/src/Js/JsExpression.php#L35, ie. `new JsExpression('let str = 'x' + [] + 'y';', ['arg'])`

> Also, a Behat (functional) test for testing the reveal functonality is needed.

see https://github.com/atk4/ui/tree/develop/tests-behat directory, create demo with password file and `password.feature` test which will check:
1. password is hidden
2. click reveal
3. password is shown
4. click reveal again
5. password must be hidden again

you can either setup a testing environment on your PC or use the CI to make it passing using trial and error

Reply:
ID - button has `->name` too, the same for inner icon probably

Reply:
It is because the `$button` is not initialized, it needs to be part of the render tree first.

Reply:
Looking into https://github.com/atk4/ui/blob/5.2.0/src/Form/Control/Input.php#L167 I belive you can replace `new Button(...)` with `Button::addTo($this, ...)`. As such call is in `$this::init()`, `addTo(` will call the init on the button and after the call the button should have the ID present.

Reply:
I will try your find, I solved it by calling $this->renderView() for now but I think your find is more elegant if it works.

Reply:
Ok, I tried it and yes, it renders the button directly but gets a tag not found error on "content"... So, I'm sticking with the working solution for now.